flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/FlutterLocalNotificationsPlugin.java

```diff
@@ -1306,4 +1317,36 @@ private void getNotificationChannels(Result result) {
         }
         return channelPayload;
     }
+
+    private void startForegroundService(MethodCall call, Result result) {
+        Map<String, Object> notificationData = call.<Map<String, Object>>argument("notificationData");
+        Integer startType = call.<Integer>argument("startType");
+        Boolean hasForegroundServiceType = call.<Boolean>argument("hasForegroundServiceType");
+        Integer foregroundServiceType = call.<Integer>argument("foregroundServiceType");
+        if (notificationData != null && startType != null && hasForegroundServiceType != null && foregroundServiceType != null) {
+            NotificationDetails notificationDetails = extractNotificationDetails(result, notificationData);
+            if (notificationDetails != null) {
+                if(notificationDetails.id!=0) {
+                    ForegroundServiceStartParameter parameter = new ForegroundServiceStartParameter(notificationDetails, startType, hasForegroundServiceType, foregroundServiceType);
+                    Intent intent = new Intent(applicationContext, ForegroundService.class);
+                    intent.putExtra(ForegroundServiceStartParameter.EXTRA, parameter);
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
I'm not sure if I get your point (haven't worked that much with AndroidX)...
The documentation you linked for `ContextCompat` states that
> startForegroundService() was introduced in O, just call startService for before O.

The way I understand this is that even with AndroidX we need to distinguish `if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O)`.
So if we need to distinguish here, too, what is the benefit of AndroidX?

Reply:
AndroidX should remove the need for having this check completely. In general, AndroidX allows for using a lot of various APIs with backwards compatibility on older Android versions. In this case, the method I mentioned would do the exact same thing as this part of the code and can be confirmed by looking at the source code.

Reply:
This means that I must have gotten the documentation of [`ContextCompat.startForegroundService`](https://developer.android.com/reference/androidx/core/content/ContextCompat#startForegroundService(android.content.Context,%20android.content.Intent)) wrong?
I understand it as command since it commands me too `just call`. If there was a third persons `s` after `call` I would have interpreted it as `it` (the method) `just calls` ...

Again, it reads that
> startForegroundService() was introduced in O, just call startService for before O.

but with respect to your comment shouldn't it be this?
> startForegroundService() was introduced in O, so ***this method*** just call***s*** startService for before O.

But thats a problem in AndroidX and not ours right now 😄. I got your point now and will just use `ContextCompat` without a version check.

Reply:
> This means that I must have gotten the documentation of ContextCompat.startForegroundService wrong?

What you wrote is probably what they intended to communicate on what it does but weren't clear on as it can be misunderstood as telling developers what they should do, especially if they're not familiar with Android X and the various compat libraries